File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> ans;
    
    ans.push_back(arr[0]);
    
    for(int x = 0; x < arr.size(); x++) if(ans[ans.size() - 1] != arr[x]) ans.push_back(arr[x]);
    
    return ans;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        List<Integer> li = new ArrayList<>();
        
        li.add(arr[0]);
        for(int i = 1; i < arr.length; i++) {
            if(li.get(li.size() - 1) == arr[i]) continue;
            li.add(arr[i]);
        }
        
        return li.stream().mapToInt(i -> i).toArray();
    }
}